### PR TITLE
refactor: remove v2 subdirectory for SDK

### PR DESCRIPTION
**Pull request by hahattan**
move application, cache package out of v2 subdirectory
remove v2HttpController and move the associated method to
internal/controller/http/RestController

Signed-off-by: Chris Hung <chris@iotechsys.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-c/blob/master/.github/CONTRIBUTING.md

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

File: pkg/service/service.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ import (
 	"github.com/edgexfoundry/device-sdk-go/v2/internal/clients"
 	"github.com/edgexfoundry/device-sdk-go/v2/internal/common"
 	"github.com/edgexfoundry/device-sdk-go/v2/internal/container"
-	"github.com/edgexfoundry/device-sdk-go/v2/internal/controller"
+	restController "github.com/edgexfoundry/device-sdk-go/v2/internal/controller/http"
```
Comment (lenny-intel) [inline pkg/service/service.go:36]: why is alias needed here? I don't see any name conflicts. Alisa should only be used to avoid name conflicts.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by lenny-intel — COMMENTED** [review]
Also, I expected to see more V1 code files deleted. Am I missing something?

File: pkg/service/service.go (hunk 1 above)
Comment (hahattan) [reply pkg/service/service.go:36]: this file imports `net/http` package, used by `AddRoutes` method. 
any idea about a better alias name ?

Comment (lenny-intel) [reply pkg/service/service.go:36]: Ohh, I missed that the name conflict is on `http`, not `controller`. I am good with this.

**Comment by lenny-intel** [comment]
@hahattan , does this eliminate all the old V1 code?

**Comment by hahattan** [comment]
> @hahattan , does this eliminate all the old V1 code?

Nah I think there are still some more files need to be removed, gonna work on it next morning.

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**